### PR TITLE
Istio Config native implementation

### PR DESCRIPTION
**Pull request by lucasponce**
This PR is a prev step for completing the links from the Service Mesh details -> Plugin.

As discussed, the Istio Config page would be implemented in a native way and the PoC only had a couple of lines to put a skeleton.

This PR would add more logic to this page:
- Introduce the standard "projects" filter as any other OpenShift Console page.
- Simplify the list of Istio resources, just pulling the basic information, but adding all types into this logic.
- Update the Overview -> Istio Config links in the Kiali -> Plugin integration jump.

This is a preliminary step, as this will enable the navigation from the "Service Mesh" details pages into these pages.

Note, that is out of the scope to bring the Kiali validation logic to these pages, that's something we'll do in the next effort, as in this case, the plugin is implementing a new page. In the next iteration, we'll study the best approach to augment this information with the Kiali validations.

This work doesn't require updates on the Kiali backend side, it's only scoped to the plugin code.

istio-config-links

### CODE REVIEW

**Review by jmazzitelli — APPROVED** [review]
This plugin integration is so good I can't see where Kiali ends and where Console begins :)
LGTM

One thing - where do I see the validation errors in the Istio Config yaml? I can't see to find it. I assume this will be coming later in future PRs.

File: plugin/src/k8s/types.ts [hunk 1]
```diff
@@ -1,165 +0,0 @@
-import { K8sResourceCommon } from '@openshift-console/dynamic-plugin-sdk';
```
[the remark this answers is not in the document]
Comment (lucasponce) [reply plugin/src/k8s/types.ts:1]: Yes. For this new implementation of the Istio Config list, we are not using a fully typed representation of the Istio objects, but a generic one, that is enough to fetch the name, namespace, and kind:
image

Then this page can query all the types reusing the common objects:
image

Also, the "detail" is not implemented by the plugin, but it's the generic page that the OpenShift Console has to edit any resource.

**Comment by lucasponce** [comment]
> One thing - where do I see the validation errors in the Istio Config yaml? I can't seem to find it. I assume this will be coming later in future PRs.

Not yet implemented, that's something to implement later, once this work is consolidated https://github.com/kiali/kiali/pull/5184.

This plugin is implemented a new IstioConfig page with the dynamic-plugin-sdk and re-using the editor pages that are builtin in OpenShift Console.

We'd need to fetch the Kiali validations and combine the results on these details (not yet defined, there are several options to consider here how to do it).

**Comment by lucasponce** [comment]
Thanks for the reviews.
I'm going to consolidate this one, it's a required step to continue with the "links" from the ServiceMesh tabs into other areas.

Now, when an Istio resource is linked in Kiali, the plugin will update the link to these pages.

**Pull request merged**